### PR TITLE
Implemented town-wide chat in Player

### PR DESCRIPTION
**Pull request by abigailhodge**
New Feature:
- Added townMessageChain to Player
- Added receiveMessage to Player
- Added tests

Refactors/Changes:
- Made message an optional variable in MessageChain construction (it's only needed when creating a direct message chain, and a player should always enter a room with an empty town and proximity MessageChain)
- Added a new PlayerData type so that Message doesn't have a cyclical dependency
- Extracted a few functions from MessageChain's test and put them in a common TestUtils file
- Linter fixups

### CODE REVIEW

File: services/roomService/src/types/MessageChain.ts [hunk 1]
```diff
@@ -1,61 +1,66 @@
...
-    }
+  private readonly _participants: string[] | undefined;
 
-    set isActive(value: boolean) {
-        this._isActive = value;
-    }
+  constructor(message?: Message) {
+    this._isActive = true;
+    if (message !== undefined) {
+      // message will only be passed in when starting a direct message chain
```
Comment (anikarabenhorst) [inline services/roomService/src/types/MessageChain.ts:20]: good catch, thank you

File: services/roomService/src/types/Player.ts [hunk 2]
```diff
@@ -23,6 +26,7 @@ export default class Player {
     };
     this._userName = userName;
     this._id = nanoid();
+    this._townMessageChain = new MessageChain;
```
Comment (michelle-pine) [inline services/roomService/src/types/Player.ts]: shouldn't this be `new MessageChain()` or is the linter removing those parens?

the parens make it more clear it's an instantiation, it's pretty weird in my mind not to have this

File: services/roomService/src/types/Player.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { MessageType } from '../CoveyTypes';
```
Comment (michelle-pine) [inline services/roomService/src/types/Player.test.ts:1]: thanks for moving these!

File: services/roomService/src/types/Player.test.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+  describe('receiveMessage', () => {
+    it('adds an incoming townwide message to the town message list', () => {
+      const player = createPlayerForTesting();
+      const message = createMessageForTesting(MessageType.TownMessage, player);
+      player.receiveMessage(message);
+      const townMessages = player.townMessageChain.messages;
+      expect(townMessages.length).toBe(1);
+      expect(townMessages[0]).toBe(message);
```
Comment (michelle-pine) [inline services/roomService/src/types/Player.test.ts]: fun fact: you can just use `expect(townMessages).toContain(message)`, especially if you already checked the length. not something you need to change but it just makes it more readable

**Review by michelle-pine — APPROVED** [review]
Looks great!

**Review by anikarabenhorst — APPROVED** [review]
lgtm

**Pull request merged**